### DISCUSSION TITLE
What is namespace for When() method while applying conditional branching in elsa 3.0

### DISCUSSION THREAD

**Discussion by NiveditaGalwade** [elsa-workflows/elsa-core#5594] [Q&A]
public class CustomWorkflow : IWorkflow
{
    public void Build(IWorkflowBuilder builder)
    {
        builder
            .StartWith<Activity1>()
            .Then<Activity2>()
            // Add a conditional activity here
            .If(context => context.GetInput<bool>("Flag"))
                .Then<Activity3>() // This will execute if the flag is true
            .Else()
                .Then<Activity4>() // This will execute if the flag is false
            .Then<Activity4>(); // Continue with Activity4 after Activity3 if the flag was true
    }
}

getting below error 
Error	CS1061	'ISetupActivity<If>' does not contain a definition for 'When' and no accessible extension method 'When' accepting a first argument of type 'ISetupActivity<If>' could be found (are you missing a using directive or an assembly reference?)

**Comment by glime-ai** [comment]
It seems like you are trying to add a conditional flow to your workflow using an `If` activity, but you're encountering an error because the `ISetupActivity<If>` interface does not contain a definition for `When`. This error suggests that the way you are trying to define the conditional branching is not correct or the method you are trying to use does not exist in the version of Elsa you are working with.

In Elsa 2.x, conditional branching is typically done using the `If` activity, but the syntax you are using does not match the expected syntax for setting up an `If` activity. Here's an example of how you might define an `If` activity in Elsa 2.x:

```csharp
public class CustomWorkflow : IWorkflow
{
    public void Build(IWorkflowBuilder builder)
    {
        builder
            .StartWith<Activity1>()
            .Then<Activity2>()
            .If(context => context.GetInput<bool>("Flag"), branch => branch
                .Then<Activity3>()) // This will execute if the flag is true
            .Else(branch => branch
                .Then<Activity4>()) // This will execute if the flag is false
            .Then<Activity5>(); // Continue with Activity5 after the conditional branch
    }
}
```

Please note that `Activity4` is used in the `Else` branch and `Activity5` is used to continue after the conditional branch. Make sure that `Activity4` is not intended to be executed twice in different branches unless that is your specific requirement.

If you are using Elsa 3.x, the syntax for defining workflows has changed significantly, and you would need to use the new API. Please refer to the Elsa 3.x documentation for the correct syntax if you are using the latest version.

If you continue to experience issues, please provide more information about the version of Elsa you are using and ensure that you have the correct using directives and references in your project.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
